packages/lodestar/src/sync/backfill/verify.ts

```diff
@@ -0,0 +1,49 @@
+import {allForks, CachedBeaconState} from "@chainsafe/lodestar-beacon-state-transition";
+import {IBeaconConfig} from "@chainsafe/lodestar-config";
+import {Root, allForks as allForkTypes, ssz} from "@chainsafe/lodestar-types";
+import {GENESIS_SLOT} from "@chainsafe/lodestar-params";
+import {IBlsVerifier} from "../../chain/bls";
+import {BackfillSyncError, BackfillSyncErrorCode} from "./errors";
+
+export function verifyBlockSequence(
+  config: IBeaconConfig,
+  blocks: allForkTypes.SignedBeaconBlock[],
+  anchorRoot: Root
+): {
+  nextAnchor: allForkTypes.SignedBeaconBlock | null;
+  verifiedBlocks: allForkTypes.SignedBeaconBlock[];
+  error?: BackfillSyncErrorCode.NOT_LINEAR;
+} {
+  let nextRoot: Root = anchorRoot;
+  let nextAnchor: allForkTypes.SignedBeaconBlock | null = null;
+
+  const verifiedBlocks: allForkTypes.SignedBeaconBlock[] = [];
+  for (const block of blocks.slice(0).reverse()) {
```

Comment:
Just loop backwards (with a regular for loop) without allocating a new array twice